tools/log_rotate/install_script.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Create directory, which keeps all logrotate-config to rotate hourly.
+mkdir -p /etc/logrotate.hourly.d
+
+# By default, only daily cron executes the logrotate which rotates the logs
+# according to the configuration in /etc/logrotate.d folder. To support the
+# hourly execution of logrotate, we have created logrotate config-files specific
+# hourly-execution.
+
+# This file describes about the hourly execution of logrotate. All the logrotate
+# -config files present in /etc/logrotate.hourly.d will be executed hourly.
+cat << EOF | tee /etc/logrotate.hourly.conf
+# This to enforce logrotate to run as root/adm. In ubuntu distributions, the
+# ownership of the file created via rsyslog is syslog hence, we need to add
+# this explicitly.
+su root adm
```

Comment:
as discussed, to confirm this doesn't cause any issues.

Reply:
By default, log-rotation is done via root user. In case of the owner of the file is 'syslog', if we don't change we get the below error info, which say to explicitly convey the user/group using which logrotate will rotate the log.


Error Info: Parent directory has insecure permissions (It's world writable or writable by group which is not "root") Set "su" directive in config file to tell log-rotate which user/group should be user for rotation.